instant-distance-py/examples/translations/translate.py

```diff
@@ -0,0 +1,142 @@
+            ) as bar:
+                async with session.get(url) as resp:
+                    while True:
+                        lineno += 1
+                        line = await resp.content.readline()
+                        if not line:
+                            # EOF
+                            break
+
+                        # We just use the top 100k embeddings to
+                        # save on space and time
+                        elif lineno > MAX_LINES:
+                            break
+                        else:
+                            linestr = line.decode("utf-8")
+                            tokens = linestr.split(" ")
+
+                            # The first token is the word and the rest
+                            # are the embedding
+                            value = tokens[0]
+                            embedding = [float(p) for p in tokens[1:]]
+
+                            # We only go from english to the other two languages
+                            if lang == "en":
+                                word_map[value] = embedding
+                            else:
+                                # We track values here to build the instant-distance index
+                                # Every value is prepended with 2 character language code.
+                                # This allows us to determine language output later.
+                                values.append(lang + value)
+                                points.append(embedding)
+
+                            bar.next()
+
+    # Build the instant-distance index and dump it out to a file with .idx suffix
+    print("Building index... (this will take a while)")
+    hnsw = instant_distance.HnswMap.build(points, values, instant_distance.Config())
+    hnsw.dump(BUILT_IDX_PATH)
+
+    # Store the mapping from string to embedding in a .json file
+    with open(WORD_MAP_PATH, "w") as f:
+        f.write(json.dumps(word_map))
+
+
+async def translate(word):
+    """
+    This function relies on the index built in the `download_build_index` function.
+    If the data does not yet exist, it will download and build the index.
+
+    The input is expected to be english. A word is first mapped onto an embedding
+    from the mapping stored as json. Then we use instant-distance to find the approximate
+    nearest neighbors to that point (embedding) in order to translate to other languages.
+    """
+    data_exists = os.path.isfile(BUILT_IDX_PATH) and os.path.isfile(WORD_MAP_PATH)
+    if not data_exists:
+        print("Instant Distance index not present. Building...")
+        await download_build_index()
+
+    print("Loading indexes from filesystem...")
+    with open(WORD_MAP_PATH, "r") as f:
+        word_map = json.loads(f.read())
+
+    # Get an embedding for the given word
+    try:
+        embedding = word_map[word]
+    except KeyError:
+        print(f"Word not recognized: {word}")
+        exit(1)
```

Comment:
I usually prefer not to use exceptions for control flow. Why not just use `word_map.get(word)`?